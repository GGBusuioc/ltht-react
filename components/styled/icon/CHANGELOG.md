# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.9](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/icon@0.1.8...@ltht-react/icon@0.1.9) (2020-04-14)

**Note:** Version bump only for package @ltht-react/icon





## [0.1.8](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/icon@0.1.7...@ltht-react/icon@0.1.8) (2020-04-09)

**Note:** Version bump only for package @ltht-react/icon





## [0.1.7](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/icon@0.1.6...@ltht-react/icon@0.1.7) (2020-04-08)

**Note:** Version bump only for package @ltht-react/icon






## [0.1.6](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/icon@0.1.5...@ltht-react/icon@0.1.6) (2020-04-03)

**Note:** Version bump only for package @ltht-react/icon





## [0.1.5](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/icon@0.1.4...@ltht-react/icon@0.1.5) (2020-03-16)

**Note:** Version bump only for package @ltht-react/icon






## [0.1.4](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/icon@0.1.3...@ltht-react/icon@0.1.4) (2020-03-12)

**Note:** Version bump only for package @ltht-react/icon





## [0.1.3](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/icon@0.1.2...@ltht-react/icon@0.1.3) (2020-03-06)

**Note:** Version bump only for package @ltht-react/icon





## [0.1.2](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/icon@0.1.1...@ltht-react/icon@0.1.2) (2020-02-27)

**Note:** Version bump only for package @ltht-react/icon





## [0.1.1](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/icon@0.1.0...@ltht-react/icon@0.1.1) (2020-02-26)

**Note:** Version bump only for package @ltht-react/icon
