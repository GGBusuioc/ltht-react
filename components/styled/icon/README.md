# Icon

<!-- STORY -->

### Import

```js
import Icon from '@ltht-react/icon'
```

### Usage

```jsx
<Icon />
```
