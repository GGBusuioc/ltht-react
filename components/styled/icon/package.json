{
  "name": "@ltht-react/icon",
  "version": "0.1.24",
  "description": "ltht-react styled Icon component.",
  "author": "LTHT",
  "homepage": "",
  "license": "MIT",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "directories": {
    "lib": "lib"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "clean": "rimraf lib",
    "build": "yarn build:lib",
    "watch": "tsc -w",
    "build:lib": "tsc",
    "type:check": "tsc --noEmit",
    "lint": "prettylint 'src/*.{ts,tsx}'"
  },
  "dependencies": {
    "@emotion/core": "^10.0.27",
    "@emotion/styled": "^10.0.27",
    "@fortawesome/fontawesome-svg-core": "^1.2.26",
    "@fortawesome/free-solid-svg-icons": "^5.12.0",
    "@fortawesome/react-fontawesome": "^0.1.8",
    "@ltht-react/styles": "^0.1.16",
    "@ltht-react/types": "^0.0.23",
    "@ltht-react/utils": "^0.1.16",
    "emotion": "^10.0.27",
    "react": "^16.12.0"
  },
  "gitHead": "d6dd6b12bacd1405494389d7c3189bbdea90886e"
}
