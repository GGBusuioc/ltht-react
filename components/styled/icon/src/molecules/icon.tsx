import { IconType, IconSize, IconStatus, IconDirection } from '@ltht-react/styles'
import { FC, HTMLAttributes } from 'react'
import {
  BullseyeIcon,
  CalendarIcon,
  CheckboxIcon,
  CheckIcon,
  ChevronCircleIcon,
  ChevronIcon,
  CircleIcon,
  ColumnIcon,
  CommentIcon,
  CounterIcon,
  CrossIcon,
  DotCircleIcon,
  EditIcon,
  EllipsisHorizontalIcon,
  EllipsisVerticalIcon,
  ExclamationIcon,
  ExternalLinkIcon,
  FilterIcon,
  FolderPlusIcon,
  GridLayoutIcon,
  HamburgerIcon,
  InfoCircleIcon,
  LayerGroupIcon,
  ListLayoutIcon,
  PauseIcon,
  PlayIcon,
  PlusIcon,
  SearchIcon,
  SpinnerIcon,
  SquareIcon,
  StopIcon,
  TableIcon,
  ToggleOffIcon,
  ToggleOnIcon,
  TrashIcon,
  UserIcon,
} from '../atoms'

const Icon: FC<IconProps> = ({ type, size, status, direction, counterValue, ...rest }) => {
  switch (type) {
    case 'bullseye': {
      return <BullseyeIcon size={size} {...rest} />
    }
    case 'calendar': {
      return <CalendarIcon size={size} {...rest} />
    }

    case 'check': {
      return <CheckIcon size={size} {...rest} />
    }

    case 'chevron-circle': {
      return <ChevronCircleIcon size={size} direction={direction ?? 'up'} {...rest} />
    }

    case 'chevron': {
      return <ChevronIcon size={size} direction={direction ?? 'up'} {...rest} />
    }

    case 'circle': {
      return <CircleIcon size={size} status={status ?? 'default'} {...rest} />
    }

    case 'column': {
      return <ColumnIcon size={size} status={status ?? 'default'} {...rest} />
    }

    case 'comment': {
      return <CommentIcon size={size} {...rest} />
    }

    case 'counter': {
      return <CounterIcon size={size} status={status ?? 'default'} value={counterValue ?? 0} {...rest} />
    }

    case 'cross': {
      return <CrossIcon size={size} {...rest} />
    }

    case 'edit': {
      return <EditIcon size={size} {...rest} />
    }

    case 'ellipsis-horizontal': {
      return <EllipsisHorizontalIcon size={size} {...rest} />
    }

    case 'ellipsis-vertical': {
      return <EllipsisVerticalIcon size={size} {...rest} />
    }

    case 'exclamation': {
      return <ExclamationIcon size={size} status={status ?? 'default'} {...rest} />
    }

    case 'external-link': {
      return <ExternalLinkIcon size={size} {...rest} />
    }

    case 'filter': {
      return <FilterIcon size={size} {...rest} />
    }

    case 'folder-plus': {
      return <FolderPlusIcon size={size} {...rest} />
    }

    case 'grid-layout': {
      return <GridLayoutIcon size={size} {...rest} />
    }

    case 'hamburger': {
      return <HamburgerIcon size={size} {...rest} />
    }

    case 'info-circle': {
      return <InfoCircleIcon size={size} status={status ?? 'default'} {...rest} />
    }

    case 'layer-group': {
      return <LayerGroupIcon size={size} {...rest} />
    }

    case 'list-layout': {
      return <ListLayoutIcon size={size} {...rest} />
    }

    case 'pause': {
      return <PauseIcon size={size} {...rest} />
    }

    case 'play': {
      return <PlayIcon size={size} {...rest} />
    }

    case 'plus': {
      return <PlusIcon size={size} {...rest} />
    }

    case 'spinner': {
      return <SpinnerIcon size={size} {...rest} />
    }

    case 'stop': {
      return <StopIcon size={size} {...rest} />
    }

    case 'table': {
      return <TableIcon size={size} status={status ?? 'default'} {...rest} />
    }

    case 'toggle-off': {
      return <ToggleOffIcon size={size} {...rest} />
    }

    case 'toggle-on': {
      return <ToggleOnIcon size={size} {...rest} />
    }

    case 'trash': {
      return <TrashIcon size={size} {...rest} />
    }

    case 'user': {
      return <UserIcon size={size} {...rest} />
    }

    case 'dot-circle': {
      return <DotCircleIcon size={size} status={status ?? 'default'} {...rest} />
    }

    case 'search': {
      return <SearchIcon size={size} {...rest} />
    }

    case 'checkbox': {
      return <CheckboxIcon size={size} {...rest} />
    }

    case 'square': {
      return <SquareIcon size={size} {...rest} />
    }

    default: {
      return <SquareIcon size={size} {...rest} />
    }
  }
}

export interface IconProps extends HTMLAttributes<SVGElement> {
  type: IconType
  size: IconSize
  status?: IconStatus
  direction?: IconDirection
  counterValue?: number
}

export default Icon
