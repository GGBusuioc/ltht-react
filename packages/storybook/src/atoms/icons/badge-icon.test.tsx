import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import { BadgeIcon } from '@ltht-react/icon'

describe('BadgeIcon', () => {
  const Icons: React.FC = () => {
    return (
      <>
        <BadgeIcon size="medium" status="secondary" />
        <BadgeIcon size="medium" status="secondary" />
        <BadgeIcon size="medium" status="secondary" />
        <BadgeIcon size="medium" status="secondary" />
      </>
    )
  }

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Icons />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<Icons />)).toMatchSnapshot('wrapper mount')
  })
})
