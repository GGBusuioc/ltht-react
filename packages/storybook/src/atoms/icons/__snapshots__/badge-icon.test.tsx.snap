// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeIcon matches snapshot: wrapper mount 1`] = `
.emotion-0 {
  color: #003087;
}

.emotion-2 {
  font-weight: bold;
}

<Icons>
  <BadgeIcon
    size="medium"
    status="secondary"
  >
    <span
      className="fa-layers fa-fw"
    >
      <i
        className="fas"
      >
        <FontAwesomeIcon
          border={false}
          className="emotion-0"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f111",
                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z",
              ],
              "iconName": "circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-circle fa-w-16 fa-2x emotion-0"
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </i>
      <span
        className="fa-layers-text fa-inverse fa-xs emotion-2"
      >
        2
      </span>
    </span>
  </BadgeIcon>
  <BadgeIcon
    size="medium"
    status="secondary"
  >
    <span
      className="fa-layers fa-fw"
    >
      <i
        className="fas"
      >
        <FontAwesomeIcon
          border={false}
          className="emotion-0"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f111",
                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z",
              ],
              "iconName": "circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-circle fa-w-16 fa-2x emotion-0"
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </i>
      <span
        className="fa-layers-text fa-inverse fa-xs emotion-2"
      >
        2
      </span>
    </span>
  </BadgeIcon>
  <BadgeIcon
    size="medium"
    status="secondary"
  >
    <span
      className="fa-layers fa-fw"
    >
      <i
        className="fas"
      >
        <FontAwesomeIcon
          border={false}
          className="emotion-0"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f111",
                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z",
              ],
              "iconName": "circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-circle fa-w-16 fa-2x emotion-0"
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </i>
      <span
        className="fa-layers-text fa-inverse fa-xs emotion-2"
      >
        2
      </span>
    </span>
  </BadgeIcon>
  <BadgeIcon
    size="medium"
    status="secondary"
  >
    <span
      className="fa-layers fa-fw"
    >
      <i
        className="fas"
      >
        <FontAwesomeIcon
          border={false}
          className="emotion-0"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f111",
                "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z",
              ],
              "iconName": "circle",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="2x"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-circle fa-w-16 fa-2x emotion-0"
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </i>
      <span
        className="fa-layers-text fa-inverse fa-xs emotion-2"
      >
        2
      </span>
    </span>
  </BadgeIcon>
</Icons>
`;
